docs/source/more_types.rst

```diff
@@ -933,3 +933,210 @@ and non-required keys, such as ``Movie`` above, will only be compatible with
 another TypedDict if all required keys in the other TypedDict are required keys in the
 first TypedDict, and all non-required keys of the other TypedDict are also non-required keys
 in the first TypedDict.
+
+Final attributes of classes and modules
+***************************************
+
+There are several situations where static guarantees about non-redefinition
+of certain names (or references in general) can be useful. One such example
+is module or class level constants, user might want to guard them against
+unintentional modifications:
+
+.. code-block:: python
+
+   RATE = 3000
+
+   class Base:
+       DEFAULT_ID = 0
+
+   # 1000 lines later
+
+   class Derived(Base):
+       DEFAULT_ID = 1  # this may be unintentional
+
+   RATE = 300  # this too
+
+Another example is where a user might want to protect certain instance
+attributes from overriding in a subclass:
+
+.. code-block:: python
+
+   import uuid
+
+   class Snowflake:
+       """An absolutely unique object in the database"""
+       def __init__(self) -> None:
+           self.id = uuid.uuid4()
+
+   # 1000 lines later
+
+   class User(Snowflake):
+       id = uuid.uuid4()  # This has valid type, but the meaning
+                          # may be wrong
+
+Some other use cases might be solved by using ``@property``, but note that both
+above use cases can't be solved this way. For such situations, one might want
+to use ``typing.Final``.
```

Comment:
This won't be in `typing` at least until a new Python version has been released, and even then older Python versions can't use this. Use a name that can be imported now (either `typing_extensions.Final` or `mypy_extensions.Final`). Once some version of Python ships `Final`, we can change this back and add a note about `typing_extensions` for users of earlier Python versions.